sql/binlog.cc

```diff
@@ -1092,16 +1100,36 @@ class Binlog_event_writer
 
       // Store end_log_pos
       int4store(*buf_p + LOG_POS_OFFSET, end_log_pos);
-    }
+      DBUG_ASSERT(output_cache == mysql_bin_log.get_log_file());
+
+      len= *event_len_p;
 
+      if (event_encrypter.ctx)
```

Comment:
Here and everywhere with similar pattern consider "if (event_encrypter.is_encryption_enabled())" or similar

Reply:
Done